libs/blocks/global-navigation/global-navigation.js

```diff
@@ -523,24 +523,30 @@ class Gnav {
 
     const onAnalyticsEvent = (data) => {
       if (!data) return;
-
       const getInteraction = () => {
-        const {
-          event: { type, subtype } = {},
-          source: { name } = {},
-          content: { name: contentName } = {},
-        } = data;
-
-        switch (`${name}|${type}|${subtype}|${contentName || ''}`) {
-          case 'profile|click|sign-in|':
+        const { workflow, type, subtype } = data.event || data;
+        const contentName = data.content?.name;
+
+        switch (`${workflow}|${type}|${subtype}${contentName ? `|${contentName}` : ''}`) {
```

Comment:
This might make the switch condition a little clearer
```suggestion
        const contentName = data.content?.name ? `|${data.content.name}` : ''

        switch (`${workflow}|${type}|${subtype}${contentName}`) {
```

Reply:
Agree. Made the change 👍